src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        list: a list of NetworkDevice objects identifying the network devices on each host
+
+    """
+    network_devices = []
+
+    ofi_info = get_ofi_info(hosts, supported, verbose)
+    ucx_info = get_ucx_info(hosts, supported, verbose)
+
+    interfaces = get_active_network_interfaces(hosts, verbose)
+    for host in hosts:
+        for interface, node_set in interfaces.items():
+            if host not in node_set:
+                continue
+            kwargs = {"host": host, "device": interface, "port": 1}
+            data_gather = {
+                "ib_device": get_interface_ib_name(node_set, interface, verbose),
+                "provider": get_interface_providers(interface, ofi_info),
+                "numa": get_interface_numa_node(node_set, interface, verbose),
+            }
+            for ib_name in data_gather["ib_device"]:
+                for add_on in ([], ["1"]):
+                    device = ":".join([ib_name] + add_on)
+                    data_gather["provider"].update(get_interface_providers(device, ucx_info))
+            for key, data in data_gather.items():
+                kwargs[key] = []
+                for item, item_node_set in data.items():
+                    if node_set == item_node_set:
+                        kwargs[key].append(item)
+                if kwargs[key]:
+                    kwargs[key] = ",".join([str(item) for item in kwargs[key]])
+                else:
+                    kwargs[key] = None
+
+            for item in kwargs["provider"].split(","):
+                these_kwargs = kwargs.copy()
+                these_kwargs["provider"] = item
+                network_devices.append(NetworkDevice(**these_kwargs))
+
+    return network_devices
+
+
+def get_dmg_network_information(dmg_network_scan):
+    """Get the network device information from the dmg network scan output.
+
+    Args:
+        dmg_network_scan (dict): the dmg network scan json command output
+
+    Raises:
+        CommandFailure: if there was an error processing the dmg network scan output
+
+    Returns:
+        list: a list of NetworkDevice objects identifying the network devices on each host
+
+    """
+    network_devices = []
+
+    try:
+        for host_fabric in dmg_network_scan["response"]["HostFabrics"].values():
+            for host in NodeSet(host_fabric["HostSet"].split(":")[0]):
+                for interface in host_fabric["HostFabric"]["Interfaces"]:
+                    network_devices.append(
+                        NetworkDevice(
+                            host, interface["Device"], None, 1, interface["Provider"],
+                            interface["NumaNode"])
+                    )
+    except KeyError as error:
+        raise CommandFailure(
+            "Error processing dmg network scan json output: {}".format(dmg_network_scan)) from error
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string